src/main/scala/edu/ie3/osmogrid/guardian/OsmoGridGuardian.scala

```diff
@@ -7,131 +7,89 @@
-      extends OsmoGridGuardianEvent
+  /* Messages, that are understood and sent */
+  sealed trait Request
 
-  def apply(): Behavior[OsmoGridGuardianEvent] = idle()
+  /** Message to initiate a grid generation run
+    *
+    * @param cfg
+    *   Configuration for the tool
+    * @param additionalListener
+    *   Addresses of additional listeners to be informed about results
+    * @param runId
+    *   Unique identifier for that generation run
+    */
+  final case class Run(
+      cfg: OsmoGridConfig,
+      additionalListener: Seq[ActorRef[ResultListener.ResultEvent]] = Seq.empty,
+      runId: UUID = UUID.randomUUID()
+  ) extends Request
 
-  private def idle(): Behavior[OsmoGridGuardianEvent] =
-    Behaviors.receive { (ctx, msg) =>
-      msg match {
-        case Run(cfg) =>
-          ctx.log.info("Initializing grid generation!")
+  /* dead watch events */
+  sealed trait Watch extends Request {
+    val runId: UUID
+  }
 
-          ctx.log.info("Starting input data provider")
-          val inputProvider =
-            ctx.spawn(InputDataProvider(cfg.input), "InputDataProvider")
-          ctx.watchWith(inputProvider, InputDataProviderDied)
-          ctx.log.debug("Starting output data listener")
-          val resultEventListener =
-            ctx.spawn(ResultListener(cfg.output), "ResultListener")
-          ctx.watchWith(resultEventListener, ResultEventListenerDied)
+  private[guardian] final case class RunGuardianDied(override val runId: UUID)
+      extends Watch
 
-          /* Check, which voltage level configs are given. Start with lv level, if this is desired for. */
-          cfg.generation match {
-            case Generation(Some(lvConfig)) =>
-              ctx.log.debug("Starting low voltage grid coordinator.")
-              val lvCoordinator = ctx.spawn(LvCoordinator(), "LvCoordinator")
-              ctx.watchWith(lvCoordinator, LvCoordinatorDied)
-              lvCoordinator ! ReqLvGrids(lvConfig, ctx.self)
-              awaitLvGrids(inputProvider, resultEventListener)
-            case unsupported =>
-              ctx.log.error(
-                "Received unsupported grid generation config. Bye, bye."
-              )
-              Behaviors.stopped
-          }
-        case InputDataProviderDied =>
-          ctx.log.error("Input data provider died. That's bad...")
-          Behaviors.stopped
-        case ResultEventListenerDied =>
-          ctx.log.error("Result event listener died. That's bad...")
-          Behaviors.stopped
-        case LvCoordinatorDied =>
-          ctx.log.error("Lv coordinator died. That's bad...")
-          Behaviors.stopped
-        case unsupported =>
-          ctx.log.error(s"Received unsupported message '$unsupported'.")
-          Behaviors.stopped
-      }
-    }
+  /** Relevant, state-independent data, the the actor needs to know
+    *
+    * @param runs
+    *   Currently active conversion runs
+    */
+  private[guardian] final case class GuardianData(
```

Comment:
Why package private? Not required, can be just private

```suggestion
  private final case class GuardianData(
```